mlvtools/export_pipeline.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+import argparse
+import glob
+import logging
+from collections import namedtuple
+from os import chmod
+from os.path import abspath, join, exists
+from os.path import realpath, dirname
+from typing import List
+
+from jinja2 import FileSystemLoader, Environment
+
+from mlvtools.cmd import CommandHelper, ArgumentBuilder
+from mlvtools.exception import MlVToolException
+from mlvtools.helper import get_git_top_dir
+from mlvtools.mlv_dvc.dvc_parser import get_dvc_dependencies
+
+ARG_IDENTIFIER = '-'
+
+logging.getLogger().setLevel(logging.INFO)
+CURRENT_DIR = realpath(dirname(__file__))
+
+PIPELINE_EXPORT_TEMPLATE_NAME = 'pipeline-export.pl'
+ConfigurableCmds = namedtuple('ConfigurableCmds', ('cmds', 'variables'))
+
+
+def write_template(output_path, template_name: str, **kwargs):
+    """
+        Write pipeline export Jinja template. The generated file is executable.
+    """
+    loader = FileSystemLoader(searchpath=join(CURRENT_DIR, '..', 'template'))
+    jinja_env = Environment(loader=loader)
+    content = jinja_env.get_template(template_name) \
+        .render(**kwargs)
+    with open(output_path, 'w') as fd:
+        fd.write(content)
+    chmod(output_path, 0o755)
+
+
+def get_dvc_files(dvc_target_file: str) -> List[str]:
+    """
+        Return the list of potential DVC meta file pipeline step.
+        DVC meta files are all located in the same directory for a given pipeline.
+        DVC file extension: .dvc
+    """
+    if not exists(dvc_target_file):
+        raise MlVToolException(f'Targeted pipeline metadata step {dvc_target_file} does not exist')
+    return glob.glob(join(dirname(dvc_target_file), '*.dvc'))
+
+
+def export_pipeline(dvc_meta_file: str, output: str, work_dir: str):
+    """
+     Generate an executable script to run a whole pipeline
+    """
+
+    ordered_dvc_metas = get_dvc_dependencies(dvc_meta_file, get_dvc_files(dvc_meta_file))
+
+    template_data = {'work_dir': work_dir, 'cmds': [dvc_meta.cmd for dvc_meta in ordered_dvc_metas]}
```

Comment:
Not linked directly to this PR, but having Flake8/Pylint/Black checks on this codebase would probably make things easier to review / maintain :)

Reply:
Flake8 is already run in the CI https://circleci.com/gh/peopledoc/ml-versioning-tools/84?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link